tree/ntuple/v7/inc/ROOT/RFieldBase.hxx

```diff
@@ -65,6 +66,7 @@ This is and can only be partially enforced through C++.
 */
 // clang-format on
 class RFieldBase {
+   friend class ROOT::Experimental::RNTupleJoinProcessor;                    // needs ConstuctValue
```

Comment:
Which comment are you referring to? In principle the use of `ConstructValue` is still a bit of a placeholder until we properly define what should happen when no entry is found in the join. In some cases we probably want to throw an exception, or in general we want to more clearly signal to the user that the join was incomplete. I've added a TODO comment to emphasize this.

Reply:
Ah, but `EmplaceNewValue` replaces the object pointer (i.e. downstream it calls `RFieldBase::CreateValue()` rather than `RFieldBase::ConstructValue()`, so that would invalidate the pointers if they are being used by the end-user..